forc-plugins/forc-crypto/src/content.rs

```diff
@@ -0,0 +1,63 @@
+use std::str::FromStr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::{
+    fs::read,
+    io::{self, Read},
+};
+
+static STDIN_HAS_BEEN_USED: AtomicBool = AtomicBool::new(false);
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Content {
+    Path(std::path::PathBuf, Vec<u8>),
+    Raw(Vec<u8>),
+}
+
+impl Content {
+    pub fn from_hex_or_raw(input: Vec<u8>) -> Vec<u8> {
+        if let Ok(text) = String::from_utf8(input.clone()) {
+            let text = text.trim();
+            if let Some(text) = text.strip_prefix("0x") {
+                if let Ok(bin) = hex::decode(text) {
+                    return bin;
+                }
+            }
+            text.as_bytes().to_vec()
+        } else {
+            input
```

Comment:
Do we really want to return the input that was passed in here? Perhaps we should be throwing an error instead..

```rust
Err(anyhow!("Invalid input"))
```

Reply:
I agree. We should accept valid UTF-8 data *or* any raw binary data encoded as hexadecimal